client/balrogclient/api.py

```diff
@@ -118,12 +119,17 @@ def do_request(self, url, data, method):
             logging.debug('Data sent: %s', sanitised_data)
         else:
             logging.debug('Data sent: %s', data)
+        data = json.dumps(data)
+        if method == 'PUT' or method == 'POST':
+            data = data.encode('utf-8')
+            data = gzip.compress(data)
         headers = {'Accept-Encoding': 'application/json',
                    'Accept': 'application/json',
-                   'Content-Type': 'application/json'}
+                   'Content-Type': 'application/json',
+                   'Content-Encoding': 'gzip'}
```

Comment:
We shouldn't set the Content-Encoding header on non-PUT/POST requests, even though it appears to be harmless. Could you rearrange the code a little so that original headers are set, then do the `if method in ("PUT", "POST"):` block, where you extend the headers and do the compression.